plugins/amoeba/platforms/cuda/src/AmoebaCudaKernels.cpp

```diff
@@ -2011,6 +2032,48 @@ void CudaCalcAmoebaMultipoleForceKernel::getInducedDipoles(ContextImpl& context,
     }
 }
 
+
+void CudaCalcAmoebaMultipoleForceKernel::getTotalDipoles(ContextImpl& context, vector<Vec3>& dipoles) {
+    ensureMultipolesValid(context);
+    int numParticles = cu.getNumAtoms();
+    dipoles.resize(numParticles);
+    const vector<int>& order = cu.getAtomIndex();
+    if (cu.getUseDoublePrecision()) {
+        vector<double4> posqVec;
+        vector<double> labDipoleVec;
+        vector<double> inducedDipoleVec;
+        double totalDipoleVecX;
+        double totalDipoleVecY;
+        double totalDipoleVecZ;
+        inducedDipole->download(inducedDipoleVec);
+        labFrameDipoles->download(labDipoleVec);
+        cu.getPosq().download(posqVec);
+        for (int i = 0; i < numParticles; i++) {
+            totalDipoleVecX = posqVec[i].x * posqVec[i].w + labDipoleVec[3*i] + inducedDipoleVec[3*i];
+            totalDipoleVecY = posqVec[i].y * posqVec[i].w + labDipoleVec[3*i+1] + inducedDipoleVec[3*i+1];
+            totalDipoleVecZ = posqVec[i].z * posqVec[i].w + labDipoleVec[3*i+2] + inducedDipoleVec[3*i+2];
```

Comment:
Hi Dr. Eastman,

I've modified the definitions of the functions in
plugins/amoeba/platforms/cuda/src/AmoebaCudaKernels.cpp
and plugins/amoeba/platforms/reference/src/SimTKReference/AmoebaReferenceMultipoleForce.cpp
and changed the test values in
plugins/amoeba/platforms/cuda/tests/TestCudaAmoebaMultipoleForce.cpp
and plugins/amoeba/platforms/reference/tests/TestReferenceAmoebaMultipoleForce.cpp

Thanks.

Saurabh

On 5 July 2016 at 11:49, peastman <notifications@github.com> wrote:

> In plugins/amoeba/platforms/cuda/src/AmoebaCudaKernels.cpp
> <https://github.com/pandegroup/openmm/pull/1534#discussion_r69617153>:
>
> > +    dipoles.resize(numParticles);
> > +    const vector<int>& order = cu.getAtomIndex();
> > +    if (cu.getUseDoublePrecision()) {
> > +        vector<double4> posqVec;
> > +        vector<double> labDipoleVec;
> > +        vector<double> inducedDipoleVec;
> > +        double totalDipoleVecX;
> > +        double totalDipoleVecY;
> > +        double totalDipoleVecZ;
> > +        inducedDipole->download(inducedDipoleVec);
> > +        labFrameDipoles->download(labDipoleVec);
> > +        cu.getPosq().download(posqVec);
> > +        for (int i = 0; i < numParticles; i++) {
> > +            totalDipoleVecX = posqVec[i].x * posqVec[i].w + labDipoleVec[3*i] + inducedDipoleVec[3*i];
> > +            totalDipoleVecY = posqVec[i].y * posqVec[i].w + labDipoleVec[3*i+1] + inducedDipoleVec[3*i+1];
> > +            totalDipoleVecZ = posqVec[i].z * posqVec[i].w + labDipoleVec[3*i+2] + inducedDipoleVec[3*i+2];
>
> Thanks! Remember to update the test cases as well.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/pandegroup/openmm/pull/1534/files/a62b6bff8ad247678dd74eadfb749c63bb896bcb#r69617153>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ANPF-ygGvg3E0J1m8F9Ywl-PJmObqB42ks5qSqc2gaJpZM4JCpXy>
> .
>